src/components/pages/auth/registration.tsx

```diff
@@ -32,9 +61,33 @@ export function Registration() {
 
   const reorderAndSetFlow = (origFlow?: RegistrationFlow) => {
     if (!origFlow) return
-    const { nodes: n } = origFlow.ui
-    origFlow.ui.nodes = [n[0], n[1], n[3], n[2], ...n.slice(4)]
-    setFlow(origFlow)
+    const reorderedNodes = origFlow.ui.nodes
+      .map(changeButtonTypeOfSSOProvider)
+      .sort((nodeA, nodeB) => {
+        const aAttributes = nodeA.attributes as UiNodeInputAttributes
+        const bAttributes = nodeB.attributes as UiNodeInputAttributes
+
+        const aIndex =
+          SortOrderInputFields[
+            aAttributes.name as keyof typeof SortOrderInputFields
+          ]
+        const bIndex =
+          SortOrderInputFields[
+            bAttributes.name as keyof typeof SortOrderInputFields
+          ]
+        if (aIndex === undefined || bIndex === undefined) {
+          // eslint-disable-next-line no-console
+          console.warn('Unexpected name attribute value found')
+          return 0
+        }
+
+        return aIndex - bIndex
+      })
+    const newFlow = {
+      ...origFlow,
+      ui: { ...origFlow.ui, nodes: reorderedNodes },
+    }
```

Comment:
basically just taste. It's  more fp but also more code.

Reply:
Oh sorry, I just meant the actually marked lines. You could also modify the original object above and then you would not need to spread and modify here.